src/wallet/scriptpubkeyman.cpp

```diff
@@ -632,11 +632,12 @@ SigningResult LegacyScriptPubKeyMan::SignMessage(const std::string& message, con
     return SigningResult::SIGNING_FAILED;
 }
 
-std::optional<PSBTError> LegacyScriptPubKeyMan::FillPSBT(PartiallySignedTransaction& psbtx, const PrecomputedTransactionData& txdata, int sighash_type, bool sign, bool bip32derivs, int* n_signed, bool finalize) const
+std::optional<PSBTError> LegacyScriptPubKeyMan::FillPSBT(PartiallySignedTransaction& psbtx, const PrecomputedTransactionData& txdata, std::optional<int> sighash_type, bool sign, bool bip32derivs, int* n_signed, bool finalize) const
 {
     if (n_signed) {
         *n_signed = 0;
     }
+    if (!sighash_type) sighash_type = SIGHASH_DEFAULT;
```

Comment:
e94bb4e5fc6dea67844a8495b948b679649b7b65: maybe use `SIGHASH_ALL` here, since the legacy wallet doesn't support Taproot.

Reply:
This is temporary anyways and is removed in a later commit.